CMakeLists.txt

```diff
@@ -83,6 +83,13 @@ IF(NOT KOKKOS_HAS_TRILINOS)
   ENDif()
   IF(NOT DEFINED ${PROJECT_NAME})
     PROJECT(Kokkos CXX)
+    IF (NOT CMAKE_BUILD_TYPE)
```

Comment:
Would this better to check `DEFINED CMAKE_BUILD_TYPE`?  There are uses for this being empty (i.e. CMake adds none of its own flags).

Reply:
The purpose of this pull request was to default to something reasonable when no build type is provided when calling `cmake`.
If I check
```
IF(NOT DEFINED CMAKE_BUILD_TYPE)
```
I still get an empty build type in that case. In other words, `CMAKE_BUILD_TYPE` is already defined (as empty) at this point.

Reply:
Yep. I see the problem. See comment below.